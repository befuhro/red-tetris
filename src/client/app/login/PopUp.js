import React from 'react';

const backgroundPopUp = {
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
};

const popUp = {
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    padding: '10px',
    backgroundColor: '#0b1d2d',
};

const errorFont = {
    color: '#61dafb',
    fontFamily: 'sans-serif',
    margin: '10px 20px 10px 20px',
};

const crossContainer = {
    alignSelf: 'flex-end',
    backgroundColor: 'white',
    height: '30px',
    width: '30px',
    borderRadius: '15px'
};

const cross = {
    fontSize: '20px',
    color: 'red',
    fontFamily: 'sans-serif',
    margin: '0'
};

function PopUp({errors, deleteErrors}) {
    return (
        <div style={backgroundPopUp}>
            <div style={popUp}>
                <button style={crossContainer}>
                    <p style={cross} onClick={deleteErrors}>x</p>
                </button>
                {errors.map((error) => <p key={error.id} style={errorFont}>{error.message}</p>)}
            </div>
        </div>
    )
}

export default PopUp;