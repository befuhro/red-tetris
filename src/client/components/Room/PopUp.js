import React from 'react';
import { Link } from 'react-router-dom';

const backgroundPopUp = {
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    position: 'fixed',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
};

const popUp = {
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    padding: '10px',
    backgroundColor: '#0b1d2d',
};

const errorFont = {
    color: '#61dafb',
    fontFamily: 'sans-serif',
    margin: '10px 20px 10px 20px',
};

const link = {
    fontFamily: 'sans-serif',
    backgroundColor: 'white',
    width: 'fit-content',
    padding: '5px',
    color: 'black',
    textDecoration: 'none',
    borderRadius: '5px',
};

function PopUp({errors}) {
    return (
        <div style={backgroundPopUp}>
            <div style={popUp}>
                {errors.map((error) => <p key={error.id} style={errorFont}>{error.message}</p>)}
                <Link style={link} to="/">Login Page</Link>
            </div>
        </div>
    )
}

export default PopUp;
