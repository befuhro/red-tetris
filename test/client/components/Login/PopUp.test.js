import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

import PopUp from  '../../../../src/client/components/Login/PopUp';

describe('<PopUp />', () => {
    let component;

   beforeAll(() => {
       component = enzyme.mount(<PopUp deleteErrors={() => {}} errors={[{id:0, message:'room is full'}]}/>);
   });

   it('basic test', () => {
       expect(component.html()).toMatchSnapshot();
   });
});
