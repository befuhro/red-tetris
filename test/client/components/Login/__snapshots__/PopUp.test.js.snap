// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUp /> basic test 1`] = `"<div style=\\"background-color: rgba(255, 255, 255, 0.5); position: fixed; display: flex; justify-content: center; align-items: center; top: 0px; left: 0px; right: 0px; bottom: 0px;\\"><div style=\\"position: absolute; display: flex; flex-direction: column; padding: 10px; background-color: rgb(11, 29, 45);\\"><button style=\\"align-self: flex-end; background-color: white; height: 30px; width: 30px; border-radius: 15px;\\"><p style=\\"font-size: 20px; color: red; font-family: sans-serif; margin: 0px;\\">x</p></button><p style=\\"color: rgb(97, 218, 251); font-family: sans-serif; margin: 10px 20px 10px 20px;\\">room is full</p></div></div>"`;
